examples/regexp-replace-operations/regexp_replace_operations.bal

```diff
@@ -0,0 +1,23 @@
+import ballerina/io;
+
+public function main() {
+    string creditCardNumber = "1234-5678-9876-5432";
+    string:RegExp pattern = re `(\d{4})-(\d{4})-(\d{4})`;
+    // Replaces the first occurrence of the credit card pattern with a masked representation.
+    string maskedCardNumber = pattern.replace(creditCardNumber, "****-****-****");
+    io:println(maskedCardNumber);
+
+    string xmlString = "<root>" +
+            "<!-- This is a comment -->" +
+            "<element1>Value 1</element1>" +
+            "<element2>Value 2</element2>" +
+            "<!-- Another comment -->" +
+            "<element3>Value 3</element3>" +
+        "</root>";
```

Comment:
Can't we use the template expressions

Reply:
Didn't use a string template here because it is going to mess up with the indentation of the output value. Output would look something like this in that case.
```
Comments removed XML: <root>
            
            <element1>Value 1</element1>
            <element2>Value 2</element2>
            
            <element3>Value 3</element3>
        </root>
```

I can also use some regex to remove these spacing and new lines. WDYT is better. Having the template or not due to these reasons.

Reply:
I would shorten the xml value to be on a single line and use an xml template expression.